api/src/main/java/org/openmrs/module/ipd/api/dao/impl/HibernateSlotDAO.java

```diff
@@ -97,4 +98,15 @@ public List<Slot> getSlotsByPatientAndVisitAndServiceType(Reference subject, Vis
 
 		return query.getResultList();
 	}
+    @Override
+    public List<Slot> getSlotsBySubjectReferenceIdAndForTheGivenTimeFrameAndServiceType(Reference subject, LocalDateTime localStartDate, LocalDateTime localEndDate) {
```

Comment:
As we not filtering or considering service type ,should we need serviceType in the method name ?

Reply:
changes have been made